src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -1701,6 +1701,17 @@ public GHPullRequestQueryBuilder queryPullRequests() {
         return new GHPullRequestQueryBuilder(this);
     }
 
+    /**
+     * Retrieves pull requests according to search terms.
+     *
+     * @param search
+     *            {@link GHPullRequestSearchBuilder}
+     * @return pull requests as the paged iterable
+     */
+    public PagedSearchIterable<GHPullRequest> searchPullRequests(GHPullRequestSearchBuilder search) {
+        return GHPullRequestSearchBuilder.from(search).repo(this).list();
+    }
+
```

Comment:
I'm not sure I understand the use case for `searchPullRequests(GHPullRequestSearchBuilder search)`, but you have a tests covering it (although, see the previous comment about passing `root`).  

I think we need at least some equivalent to `GitHub.searchPullRequests()`.  

```suggestion
    /**
     * Retrieves pull requests according to search terms.
     *
     * @param search
     *            {@link GHPullRequestSearchBuilder}
     * @return pull requests as the paged iterable
     */
    public PagedSearchIterable<GHPullRequest> searchPullRequests(GHPullRequestSearchBuilder search) {
        return GHPullRequestSearchBuilder.from(search).repo(this).list();
    }

    /**
     * Search for pull requests.
     *
     * @return gh pull request search builder
     */
    public GHPullRequestSearchBuilder searchPullRequests() {
        return root().searchPullRequests().repo(this);
    }

```

Reply:
> I'm not sure I understand the use case for `searchPullRequests(GHPullRequestSearchBuilder search)`, but you have a tests covering it (although, see the previous comment about passing `root`).
> 
> I think we need at least some equivalent to `GitHub.searchPullRequests()`.

The main idea was to use  `GHPullRequestSearchBuilder` from repository level and from global level (actually what can be done in github UI). That's why it's used in 2 places and from repository level source search is appended with current repository not to mess up results.

So you suggest to modify:
```
  public PagedSearchIterable<GHPullRequest> searchPullRequests(GHPullRequestSearchBuilder search) {
        return GHPullRequestSearchBuilder.from(search).repo(this).list();
    }
```
to smth like:

```
  public GHPullRequestSearchBuilder searchPullRequests() {
        return GHPullRequestSearchBuilder.from(this).list();
    }
```
like 
```
 GHPullRequestQueryBuilder(GHRepository repo) {
        super(repo.root());
        this.repo = repo;
    }
```
So we can have 2 constructors - 1 for global search from Github.class and 1 from GHRepository.
Then `GHPullRequestSearchBuilder#from` is not needed at all.